Sources/SwiftRaft/Log/AnyLog.swift

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: MIT
+// Copyright © 2021 makadaw
+
+
+/// Base class for Log type eraser to make Raft non-generic
+class BaseLog {
+
+    var logLastIndex: UInt {
+        fatalError("Abstract class")
+    }
+
+    var metadata: LogMetadata {
+        fatalError("Abstract class")
+    }
+
+}
+
+class AnyLog<ApplicationLog>: BaseLog where ApplicationLog: Log {
+    private var log: ApplicationLog
+
+    init(log: ApplicationLog) {
+        self.log = log
+    }
+
+    override var logLastIndex: UInt {
+        log.logLastIndex
+    }
+
+    override var metadata: LogMetadata {
+        log.metadata
+    }
+}
```

Comment:
This pattern seems a bit weird, why the base class?

Can this be expressed using protocols and the anyLog be a struct?

Reply:
Ah this is meant to be a type eraser... then this also can be a `struct` and the BaseLog could be a protocol IMO

Reply:
Yes, I left it as-is for now. Keep Raft type not generic for now. As with migration to the March snapshot, I found a compiler assert when using generics and `ConcurrentValue`. Try to reproduce it in a separate project and create an issue.

I think that the Raft type should become generic in the end, as this will make sure that the consumer adds only entries of the same type. So this code will be deleted.

Reply:
Have created a PR to remove it https://github.com/makadaw/swift-raft/pull/9